meson.build

```diff
@@ -0,0 +1,54 @@
+project(
+  'edlib',
+  'cpp', 'c',
+  version : '1.2.6',
+  default_options : [
+    'buildtype=release',
+    'warning_level=3',
+    'cpp_std=c++17',
```

Comment:
I personally don't care, but would say C++17 is a bit too much for edlib, given how hard it tries to stay on C++11-only. I would personally go for C++11 instead.

Reply:
Yes, C++11 sounds reasonable.
What do you mean exactly by edlib trying hard to stay on C++11-only :D? I am actually not against modernizing, it is just that my C++ knowledge has not been updated in the last couple of years. I am not even sure at the moment what is the big difference between C++11 and 17, would have to look it up. What is your thought on that, would you push edlib more toward C++17 in the future? And what does that mean actually?

Reply:
That wasn't meant as a bash. Going to more recent versions of C++ is a tradeoff: you can use features that make code simpler and cleaner (e.g. `std::make_unique` in C++14), but you're constraining your "target audience". By how much is debatable, but you will have the odd person asking you whether dropping from C++17 to C++14 would be possible, because they're stuck on some old MSVC or such.

If you don't mind the latter part, I would definitely start moving up to C++14 (or even 17), if not just for `std::make_unique`. Going to C++14 will generally exclude vanilla RHEL 7 users, because GCC 4.8's C++14 is pretty much non-existent or broken. Going to C++17 should be fine for RHEL 8, but Debian stable might complain.

Reply:
@SoapZA I know it wasn't a bash and I didn't take it as one! I am genuinely interested in upgrading Edlib so that is why I am asking for more info.

Sounds good, C++14 sounds like a safe bet then, let's go with it! If somebody is really using older standard, they can compile edlib with c++14 and their code with their standard and then link them together, if I am correct.

Reply:
I agree with Kaiser SoapZA. (And thanks for the review.) I shouldn't have specified `C++17`. However, I noticed that you were moving to pure C++ on a branch, so it's a tough decision. As a library, the more old compatibility, the better; as a project, it's definitely more fun to take advantage of newer language features. I hope to use C++20 myself soon.

Reply:
@cdunn2001 that is correct, I realized that not many people seem to be using C version (although I am not 100% on that) and decided that we should be C++ first, and then possibly have C bindings if needed. But yes, let's keep to C++14 then for now, for compatibility reasons, since this is a simple library.